lib/plugins/inspec-parallel/lib/inspec-parallel/validator.rb

```diff
@@ -101,6 +106,47 @@ def check_for_reporter_options(invocation_data)
         end
       end
 
+      def check_reporter_options_in_config(invocation_data)
+        config_opts = invocation_data[:thor_opts]["config"] || invocation_data[:thor_opts]["json_config"]
+        cfg_io = check_for_piped_config_from_stdin(config_opts)
+
+        if cfg_io == STDIN
+          # Scenario of using config from STDIN
+          @config_content ||= cfg_io.read
+        else
+          if config_opts.nil?
+            # Scenario of using default config.json file when path not provided
+            default_path = File.join(Inspec.config_dir, "config.json")
+            config_opts = default_path
+            return unless File.exist?(config_opts)
+          elsif !File.exist?(config_opts)
+            invocation_data[:validation_errors] << "Could not read configuration file at #{config_opts}"
+            return
+          end
+          @config_content = File.open(config_opts).read
+        end
+
+        reporter_config = JSON.parse(config_content)["reporter"] unless config_content.nil? || config_content.empty?
+        if reporter_config
+          reporter_config.each do |reporter, config|
+            unless config["stdout"] == false
```

Comment:
Why do we insist on this painful requirement? Can't we just assume all reporters that have a filename are not outputting to stdout?

Reply:
Also, why is this logic duplicated way over here in InSpec Parallel?

Reply:
> Why do we insist on this painful requirement? Can't we just assume all reporters that have a filename are not outputting to stdout?

It is possible for configurations to not have a file path, in case we are configuring it with automate. But stdout false is basic for every configuration which is either redirecting it to a file or to automate.

Reply:
> Also, why is this logic duplicated way over here in InSpec Parallel?

Yes, it is a logic fetched from `config.rb` to here to make this work, because it could not be reused from the config class itself. Config class expects inputs in different formats and so do the functions defined in that class. While here in the parallel validator class, we only have a config file path.